packages/gatsby-theme-newrelic/src/components/MDX.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { MDXRenderer } from 'gatsby-plugin-mdx';
+import { MDXProvider } from '@mdx-js/react';
+import Button from './Button';
+import MDXCallout from './MDXCallout';
+import Collapser from './Collapser';
+import Icon from './Icon';
+import InlineCode from './InlineCode';
+import Link from './Link';
+import MDXCodeBlock from './MDXCodeBlock';
+import MDXCollapserGroup from './MDXCollapserGroup';
+import MDXLink from './MDXLink';
+import MDXTable from './MDXTable';
+import MDXVideo from './MDXVideo';
+
+const defaultComponents = {
+  a: MDXLink,
+  code: MDXCodeBlock,
+  pre: (props) => props.children,
+  Button,
+  ButtonLink: (props) => <Button {...props} as={Link} />,
+  Callout: MDXCallout,
+  Collapser,
+  CollapserGroup: MDXCollapserGroup,
+  Icon,
+  InlineCode,
+  Link,
+  table: MDXTable,
+  Video: MDXVideo,
+};
+
+const MDX = ({ body, components }) => {
+  return (
+    <MDXProvider components={{ ...defaultComponents, ...components }}>
+      <MDXRenderer>{body}</MDXRenderer>
+    </MDXProvider>
+  );
+};
+
+MDX.propTypes = {
+  body: PropTypes.string.isRequired,
+  components: PropTypes.object,
```

Comment:
Out of curiosity, is `components` open-ended, or do we only support specific components?

Reply:
`components` are open ended. This is just to provide additional shortcodes needed for MDX documents in the specific sites. For example, we have a `LandingPageTile` shortcode on the docs site, which the theme does not know about. This prop gives us the ability to map that on the docs end and merge with these default components.